test/export/StructureDefinition.ExtensionExporter.test.ts

```diff
@@ -216,4 +221,415 @@ describe('ExtensionExporter', () => {
+      const extension = new Extension('MyExtension');
+      extension.contexts = [
+        {
+          value: 'MyVeryComplexExtension#extension[foo].extension[bigFoo]',
+          isQuoted: false
+        }
+      ];
+      doc.extensions.set(extension.name, extension);
+      const exported = exporter.exportStructDef(extension);
+      expect(exported.context).toEqual([
+        {
+          type: 'extension',
+          expression: 'http://example.org/StructureDefinition/mvc-extension#foo.bigFoo'
+        }
+      ]);
+    });
+
+    it('should set extension context with type "element" when the path is a deep part of a complex extension, but contains non-extension elements', () => {
+      const extension = new Extension('MyExtension');
+      extension.contexts = [
+        {
+          value: 'MyVeryComplexExtension#extension[bar].value[x].extension[secretBar]',
+          isQuoted: false
+        }
+      ];
+      doc.extensions.set(extension.name, extension);
+      const exported = exporter.exportStructDef(extension);
+      expect(exported.context).toEqual([
+        {
+          type: 'element',
+          expression:
+            'http://example.org/StructureDefinition/mvc-extension#Extension.extension:bar.value[x].extension:secretBar'
+        }
+      ]);
+    });
+
+    it('should set extension context when an alias is used for a resource URL', () => {
+      doc.aliases.set('$PROF', 'http://hl7.org/fhir/StructureDefinition/patient-proficiency');
+      const extension = new Extension('MyExtension');
+      extension.contexts = [
+        {
+          value: '$PROF#extension[level]',
+          isQuoted: false
+        }
+      ];
+      doc.extensions.set(extension.name, extension);
+      const exported = exporter.exportStructDef(extension);
+      expect(exported.context).toEqual([
+        {
+          type: 'extension',
+          expression: 'http://hl7.org/fhir/StructureDefinition/patient-proficiency#level'
+        }
+      ]);
+    });
+
+    it('should log an error when no extension or resource can be found with the provided value', () => {
+      const extension = new Extension('MyExtension');
+      extension.contexts = [
+        {
+          value: 'MysteryExtension',
+          isQuoted: false,
+          sourceInfo: {
+            file: 'Context.fsh',
+            location: {
+              startLine: 8,
+              startColumn: 7,
+              endLine: 8,
+              endColumn: 25
+            }
+          }
+        }
+      ];
+      doc.extensions.set(extension.name, extension);
+      const exported = exporter.exportStructDef(extension);
+      expect(exported.context).toBeUndefined();
+      expect(loggerSpy.getLastMessage('error')).toMatch(
+        'Could not find resource MysteryExtension to use as extension context.'
+      );
+      expect(loggerSpy.getLastMessage('error')).toMatch(/File: Context\.fsh.*Line: 8\D*/s);
+    });
```

Comment:
The last test in the custom resource block reminded me of possibly another test case that might be useful in this section. Is it worthwhile also testing that an invalid FSH path on a resource? I think the error will be triggered by the same logic, so it might be a duplicate test, so curious what you think.

Reply:
Do you think adding that type of test would be redundant? I'm still just curious to hear what you think.